### ISSUE TITLE
Immutability?

### ISSUE THREAD

**Issue by eccentric-j** [bastibe/org-journal#265]
**Is your feature request related to a problem? Please describe.**
Currently, TODO items are treated as immutable tasks. When you enable carryover and create a new journal entry, TODO items are removed from previous docs and added into the new entry. History is changed.

**Describe the solution you'd like**
Ideally, I would love it if an (attribute? or property?) could be added to the TODO items from previous entries to filter them out from org-agenda while creating a new COPY of any active TODO items in the new entry. The value would be being able to look at any day and see what was worked on, what was not completed, and what got done.

Additionally, this opens some doors for analysis being able to track how long it took to complete a task as well as making comparisons between days to summarize changes.

**Describe alternatives you've considered**
- Fork org-journal. It could be done but it may become a big project. Plus perhaps others would enjoy seeing a log of activity as well?
- Use org-journal but override\advise functions as necessary. May get messy or complex, but might be able to make a supporting package for it.
- Create a PR that creates an immutable config setting. Likely the best option.
- Create a separate web app that uses Dropbox API to generate org files in my org directory. It would allow for better tracking of tasks over days. WIll likely be a large project, though tempting as a product.

What are your thoughts on this? Do you see any value in a feature like this?

**Comment by cslux** [comment]
> Ideally, I would love it if an (attribute? or property?) could be added to the TODO items from previous entries to filter them out from org-agenda
> while creating a new COPY of any active TODO items in the new entry. The value would be being able to look at any day and see what was worked
> on, what was not completed, and what got done.

There is an open feature request about this topic, see
https://github.com/bastibe/org-journal/issues/248. It's on the 2.2.0
milestone. We would appreciate a little help.

> Additionally, this opens some doors for analysis being able to track how long it took to complete a task as well as making comparisons between
> days to summarize changes.

Indeed!

> Describe alternatives you've considered
>
> * Fork org-journal. It could be done but it may become a big project. Plus perhaps others would enjoy seeing a log of activity as well?

:-(

> * Use org-journal but override\advise functions as necessary. May get messy or complex, but might be able to make a supporting package for it.

It will be for sure a little messy and would break easily when something
changes in org-journal.

> * Create a PR that creates an immutable config setting. Likely the best option.

:-) Definitely the best option.

> * Create a separate web app that uses Dropbox API to generate org files in my org directory. It would allow for better tracking of tasks over days.
>  WIll likely be a large project, though tempting as a product.

Hmm, okay :-) Sounds interesting.

**Comment by eccentric-j** [comment]
😅 I may have got a bit carried away with my thinking-out-loud but I appreciate your time and feedback.

At some point, I do want to make a journaling\scheduling app to replace bullet journaling for me but for now, creating a PR for org-journal to add immutability feels like a big step in that direction.

I'll take a stab at a PR for this once I get the fix for carryover date selection fixed.

There's definitely a lot of similarities with #248, but I feel if we support immutability and design some functions\hooks for operating on past entries then the requested functionality could be derived. Alternatively, a separate function could be written to create the links or display them in a list. Either way, it could give users a few options for that.

Generating\using IDs could be useful but I don't feel that's a hard requirement for this project.

**Comment by eccentric-j** [comment]
I'm not sure what to do about this issue. I found myself switching back to paper-based bullet journaling, the picture below for reference. Unfortunately, even the best org-journal setup vids and tutorials I've found don't really come close to replacing what I really want. Lastly, in terms of digitizing my journaling, I think making a web app would be best due to integration with services like Google cal as well as being able to be editable from a mobile device.

![eccentric-j-sept-bujo](https://user-images.githubusercontent.com/590297/92296098-e2a83600-eefe-11ea-94fc-63c6521858bd.jpg)

Adding immutability to org-journal might be a lot of heft for the tool that a lot of people may not like or ever use. Plus it would likely require a good amount of tweaking to other third-party plugins like org-agenda to make it usable with immutability.

If people find this and request it as well then perhaps it's worth it but I'm not sure it's the most practical or useful idea right now.
